exonum/src/runtime/rust/tests.rs

```diff
@@ -51,7 +51,12 @@ fn create_block(blockchain: &BlockchainMut) -> Fork {
     Fork::from(patch)
 }
 
-fn commit_block(blockchain: &mut BlockchainMut, fork: Fork) {
+fn commit_block(blockchain: &mut BlockchainMut, mut fork: Fork) {
+    // FIXME: Due to during the `create_patch` `before_commit` hook invokes without changes in
+    // instances and artifacts, do this call again to mark pending artifacts and instances as
+    // active. [ECR-3222]
+    blockchain.dispatcher().activate_pending_entities(&mut fork);
```

Comment:
Maybe, the following interface would make more sense instead of `create_block` (**warning:** I didn't compile / test this approach):

```rust
fn create_block(blockchain: &mut BlockchainMut, action_fn: impl FnOnce(&mut Fork)) -> Patch {
    let mut fork = blockchain.fork();
    action_fn(&mut fork);
    let height = CoreSchema::new(&fork).height();
    blockchain.merge(fork.into_patch());
    let (_, patch) =
        blockchain.create_patch(ValidatorId(0), height.next(), &[], &mut BTreeMap::new());
    patch
}
```

Then, it could be used like

```rust
let patch = create_block(&mut blockchain, |fork| {
    Dispatcher::commit_artifact(fork, artifact.clone(), vec![]).unwrap();
});
commit_block(&mut blockchain, patch);
```

Reply:
To use this approach we have to redesign `create_patch` method in Blockchain because we should have possibility to revert changes made by the `action_fn`.

Thus I think that we have to fix this issue later by the separate PR.